JetMETCorrections/Type1MET/plugins/CorrectedPFMETProducer.cc

```diff
@@ -1,46 +1,70 @@
+// -*- C++ -*-
 
-/** \class CorrectedPFMETProducer
- *
- * Instantiate CorrectedMETProducer template for PFMET
- *
- * NOTE: This file also defines concrete implementation of CorrectedMETFactory template
- *       specific to PFMET
- *
- *
- * \authors Michael Schmitt, Richard Cavanaugh, The University of Florida
- *          Florent Lacroix, University of Illinois at Chicago
- *          Christian Veelken, LLR
- *
- *
- *
- */
+//____________________________________________________________________________||
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "FWCore/Utilities/interface/InputTag.h"
 
 #include "DataFormats/METReco/interface/PFMET.h"
+#include "DataFormats/METReco/interface/CorrMETData.h"
+
+#include "JetMETCorrections/Type1MET/interface/AddCorrectionsToGenericMET.h"
 
-#include "JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h"
+#include <vector>
 
-namespace CorrectedMETProducer_namespace
+//____________________________________________________________________________||
+class CorrectedPFMETProducer : public edm::EDProducer
```

Comment:
this better be thread-aware module.
From the implementation, it looks like ```edm::stream::EDProducer<> ``` is appropriate